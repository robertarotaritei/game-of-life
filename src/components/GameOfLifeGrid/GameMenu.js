import React from 'react';
import PlayArrowIcon from '@material-ui/icons/PlayArrow';
import PauseIcon from '@material-ui/icons/Pause';
import RestoreIcon from '@material-ui/icons/Restore';
import ClearIcon from '@material-ui/icons/Clear';
import AcUnitIcon from '@material-ui/icons/AcUnit';
import WhatshotIcon from '@material-ui/icons/Whatshot';
import SpaIcon from '@material-ui/icons/Spa';
import SaveIcon from '@material-ui/icons/Save';
import SpeedIcon from '@material-ui/icons/Speed';

class GameMenu extends React.Component {
	constructor() {
		super();

		this.state = {
			speed: false
		}
	}

	changeSpeed = (e) => {
		e.preventDefault();
		this.setState({ speed: !this.state.speed });
	}

	renderPlayPause = () => {
		switch (this.props.playState) {
			case 'play':
				return <button className='btnIcon'  data-testid='play' onClick={this.props.playButton} >
					<PlayArrowIcon />
				</button>;
			case 'resume':
				return <button className='btnIcon' data-testid='resume' onClick={this.props.resumeButton} >
					<PlayArrowIcon />
				</button>
			case 'pause':
				return <button className='btnIcon' data-testid='pause' onClick={this.props.pauseButton}>
					<PauseIcon />
				</button>
			default :
				return null;
		}
	}

	render() {
		return (
			<div style={{display: 'flex', verticalAlign: 'middle', paddingBottom: '10px'}}>
					{this.renderPlayPause()}
					<button className='btnIcon' data-testid='speed' onClick={this.changeSpeed}>
						<SpeedIcon />
					</button>
					{this.state.speed ? (
						<button className='btnIcon' data-testid='slow' onClick={this.props.slow} style={{ backgroundColor: '#063a4c41' }} >
							<AcUnitIcon />
						</button>
					) : null
					}
					{this.state.speed ? (
						<button className='btnIcon' data-testid='fast' onClick={this.props.fast} style={{ backgroundColor: '#063a4c41' }} >
							<WhatshotIcon />
						</button>
					) : null
					}
					<button className='btnIcon' data-testid='back' onClick={this.props.stop}>
						<RestoreIcon />
					</button>
					{this.props.history ?
						null : (
							<button className='btnIcon' data-testid='reset' onClick={this.props.clear}>
								<ClearIcon />
							</button>
						)
					}
					{this.props.history ?
						null : (
							<button className='btnIcon' data-testid='seed' onClick={this.props.seed}>
								<SpaIcon />
							</button>
						)
					}
					{this.props.history ?
						null : (
							<button className='btnIcon' data-testid='save' onClick={this.props.save}>
								<SaveIcon style={{color: this.props.loggedIn ? '#17C5FA' : '#036C8D'}}/>
							</button>
						)
					}
				{this.props.saveText !== '' ? <div><p className='textAnimationSave' style={{paddingBottom: '10px', color: !this.props.loggedIn ? 'red' : null}}>&nbsp;&nbsp;{this.props.saveText}</p></div> : null}

				{this.props.gameInformation !== '' ? <div><p className='textAnimationSave' style={{paddingBottom: '10px'}}>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;{this.props.gameInformation}</p></div> : null}
			</div>
		)
	}
}

export default GameMenu;