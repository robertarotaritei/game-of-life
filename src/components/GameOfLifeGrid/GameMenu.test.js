import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import GameMenu from './GameMenu';
import { shallow } from "enzyme";

test("GameMenu renders without crashing", () => {
  shallow(<GameMenu />);
});

it('GameMenu renders correctly', () => {
  const tree = renderer
    .create(<GameMenu />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});