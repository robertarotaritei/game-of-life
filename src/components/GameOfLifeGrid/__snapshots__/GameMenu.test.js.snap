// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameMenu renders correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "paddingBottom": "10px",
      "verticalAlign": "middle",
    }
  }
>
  <button
    className="btnIcon"
    data-testid="speed"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20.38 8.57l-1.23 1.85a8 8 0 01-.22 7.58H5.07A8 8 0 0115.58 6.85l1.85-1.23A10 10 0 003.35 19a2 2 0 001.72 1h13.85a2 2 0 001.74-1 10 10 0 00-.27-10.44zm-9.79 6.84a2 2 0 002.83 0l5.66-8.49-8.49 5.66a2 2 0 000 2.83z"
      />
    </svg>
  </button>
  <button
    className="btnIcon"
    data-testid="back"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9zm-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8H12z"
      />
    </svg>
  </button>
  <button
    className="btnIcon"
    data-testid="reset"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
    </svg>
  </button>
  <button
    className="btnIcon"
    data-testid="seed"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M8.55 12c-1.07-.71-2.25-1.27-3.53-1.61 1.28.34 2.46.9 3.53 1.61zm10.43-1.61c-1.29.34-2.49.91-3.57 1.64 1.08-.73 2.28-1.3 3.57-1.64z"
      />
      <path
        d="M15.49 9.63c-.18-2.79-1.31-5.51-3.43-7.63-2.14 2.14-3.32 4.86-3.55 7.63 1.28.68 2.46 1.56 3.49 2.63 1.03-1.06 2.21-1.94 3.49-2.63zm-6.5 2.65c-.14-.1-.3-.19-.45-.29.15.11.31.19.45.29zm6.42-.25c-.13.09-.27.16-.4.26.13-.1.27-.17.4-.26zM12 15.45C9.85 12.17 6.18 10 2 10c0 5.32 3.36 9.82 8.03 11.49.63.23 1.29.4 1.97.51.68-.12 1.33-.29 1.97-.51C18.64 19.82 22 15.32 22 10c-4.18 0-7.85 2.17-10 5.45z"
      />
    </svg>
  </button>
  <button
    className="btnIcon"
    data-testid="save"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      style={
        Object {
          "color": "#036C8D",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
      />
    </svg>
  </button>
  <div>
    <p
      className="textAnimationSave"
      style={
        Object {
          "color": "red",
          "paddingBottom": "10px",
        }
      }
    >
        
    </p>
  </div>
  <div>
    <p
      className="textAnimationSave"
      style={
        Object {
          "paddingBottom": "10px",
        }
      }
    >
           
    </p>
  </div>
</div>
`;
